### PR TITLE
Fix: reply 235 when client auth success, according to rfc2554

### PR DESCRIPTION
**Pull request by morya**
Fix: reply 235 when client auth success, according to rfc2554

### CODE REVIEW

File: smtp/server.go [hunk 1]
```diff
@@ -31,6 +31,7 @@ var SendAsSubject = regexp.MustCompile(`(?i)\[sendas:\s*([a-zA-Z0-9\.\-_]+)\]`)
 
 var (
 	ReplyOK               = ReplyLine{250, "OK"}
+	ReplyAuthOk           = ReplyLine{235, "auth success"}
```
Comment (rsesek) [inline smtp/server.go]: Can you name this `ReplyAuthOK` (capital K) to match `ReplyOK`?

**Review by rsesek — CHANGES_REQUESTED** [review]
Thank you for the patch! This is indeed non-spec-compliant so and should use the right response code.

The CI checks indicate that some more tests need to be updated.

File: smtp/server.go (hunk 1 above)
Comment (morya) [reply smtp/server.go]: of cause

**Comment by morya** [comment]
I didn't figured out how these tests are organized, will fix them later.

**Comment by rsesek** [comment]
LGTM, thanks again!

**Pull request merged**